wordpress-mobile/WordPress-Android
# Merge login lib changes and implement new interface methods

This PR merges the `woocommerce-android` changes to the login library from [this PR](https://github.com/woocommerce/woocommerce-android/pull/1150). There were several new interface methods added existing classes:

**LoginAnalyticsListener**
- `trackConnectedSiteInfoRequested`
- `trackConnectedSiteInfoFailed`
- `trackConnectedSiteInfoSucceeded`

**LoginListener**
- `showHelpFindingConnectedEmail`
- `gotConnectedSiteInfo`

I've implemented these new interface methods in `LoginActivity` and `LoginAnalyticsTracker` but there is no logic needed since none of them are currently being used in WPAndroid. Those are the only two changes to WPAndroid directly. 

Full documentation around the changes to login library are documented in [this PR](https://github.com/woocommerce/woocommerce-android/pull/1150), but the **WPAndroid app should see no difference in the login experience**.

### Recommended Test Scenarios
- Verify login flow is unchanged from the current login flow of WPAndroid
- Verify the text in each of the login views is also unchanged
- Login with email
- Login by site url
- Login w/ Magic Link
- Login w/ Google

**Note: This PR is not ready for merge. [This PR](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/22) must first be merged, and then a new branch/PR will be submitted with the fresh changes ready for final review.**
